R/edit-spreadsheets.R

```diff
@@ -0,0 +1,173 @@
+#' Create a new spreadsheet
+#' 
+#' Create a new (empty) spreadsheet in your Google Drive. The new spreadsheet
+#' will contain 1 default worksheet titled "Sheet1".
+#' 
+#' @param title the title for the new spreadsheet
+#' 
+#' @export
+new_ss <- function(title)
+{
+  dat <- data.frame("title" = title, 
+                    "mimeType" = "application/vnd.google-apps.spreadsheet")
+  
+  the_body <- jsonlite::toJSON(dat) %>% stringr::str_sub(., 2, -2)
+  
+  the_url <- "https://www.googleapis.com/drive/v2/files"
+  
+  gsheets_POST(the_url, the_body)
+  
+  message(paste0('Spreadsheet "', title, '" created in Google Drive.'))
+}
+
+
+#' Move a spreadsheet to trash{"__src__":"file","ids":"1PmCRtmcX4r7LOS-XVw6u8f-E3sVnFgamJaKCmVF3Sv0","parentId":"0AFlKxBjVWN80Uk9PVA"}
+#' 
+#' Move a spreadsheet to trash in Google Drive.
+#' 
+#' @param x the title or key of a spreadsheet
+#' 
+#' @note Using the key is useful when there are spreadsheets with the same name 
+#' since the default is to send the most recent spreadsheet to the trash.
+#' 
+#' @note Shared spreadsheets can not be removed from your Google Drive with 
+#' this function. You must remove it manually in the web browser.
+#' 
+#' @export
+delete_ss <- function(x)
+{
+  ssfeed_df <- list_spreadsheets()
+  
+  if(x %in% ssfeed_df$sheet_key) {
+    key = x
+  } else {
+    key_index <- match(x, ssfeed_df$sheet_title)
+    key <- ssfeed_df$sheet_key[key_index]
+  }
+  
+  url <- "https://www.googleapis.com/drive/v2/files"
+  
+  the_url <- slaste(url, key, "trash")
+  
+  gsheets_POST(the_url, the_body = NULL)
+  
+  message('Spreadsheet moved to trash in Google Drive.')
+}
+
+
+#' Make a copy of an existing spreadsheet
+#' 
+#' You can either make a copy of your own spreadsheet or another user's 
+#' spreadsheet. If the spreadsheet you want to make a copy of already exists in 
+#' your Google Drive, pass in the title of the spreadsheet. To get a copy of 
+#' another user's spreadsheet, enter the key of that spreadsheet. Make sure 
+#' that the target spreadsheet is made 'accessible' in the sharing dialog 
```

Comment:
I think "source" might be more clear, as "target" here could arguably refer to the thing you're writing or creating, i.e. the copy. If you agree, use "source" in the second sentence as well.